packages/fiori/src/Bar.js

```diff
@@ -133,6 +133,34 @@ class Bar extends UI5Element {
 			"label": this.design,
 		};
 	}
+
+	resizeHandler(entry) {
+		const bar = entry[0].target;
+		let changeFlex = false;
+		const barWidth = bar.offsetWidth;
+
+		for (let i = 0; i < bar.childElementCount; i++) {
+			if (barWidth / 3 < bar.children[i].offsetWidth) {
+				changeFlex = true;
+			}
+		}
+
+		if (changeFlex) {
```

Comment:
It would be best to not set dynamically classes like this. It's always better to use the framework, unless you specifically need to set the classes synchronously (and not wait for the framework's rendering cycle).

Please try the following:
 - In the resize handler only set some state of the Bar, do not change any classes. F.e. create a new metadata property that denotes that you have to apply the flex design. Change this property here.
 - In the `.hbs` file set classes to the 3 children of the bar, that depend on this state. Use `get classes()` and assign the classes in the template normally.
 - Then when the resize occurs, you can check the width and set the new property to true/false. This will invalidate the bar, and the `get classes` will get executed and then the classes will have different values due to the change of the private property, and thus you'll set the classes to the 3 divs via the framework.